src/blockchain/Blockchain.ts

```diff
@@ -13,8 +13,17 @@ import { internal } from "../utils/message";
 
 const LT_ALIGN = 1000000n
 
+export type BlockchainTransaction = Transaction & {
```

Comment:
nit: would it make sense to treat this as `SmartContractTransaction & {...}`?

Reply:
maybe not - right now the fields coincide, but later on they may carry other fields/context that is not supposed to be exposed to the user of Blockchain or may be exposed in a different way